e2etest/bgptests/unnummbered.go

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package bgptests
+
+import (
+	"context"
+	"fmt"
+	"net/netip"
+	"time"
+
+	"go.universe.tf/e2etest/pkg/config"
+	"go.universe.tf/e2etest/pkg/container"
+	"go.universe.tf/e2etest/pkg/executor"
+	"go.universe.tf/e2etest/pkg/frr"
+	frrconfig "go.universe.tf/e2etest/pkg/frr/config"
+	frrcontainer "go.universe.tf/e2etest/pkg/frr/container"
+	jigservice "go.universe.tf/e2etest/pkg/jigservice"
+	"go.universe.tf/e2etest/pkg/netdev"
+	testservice "go.universe.tf/e2etest/pkg/service"
+
+	"go.universe.tf/e2etest/pkg/k8s"
+	"go.universe.tf/e2etest/pkg/k8sclient"
+	"go.universe.tf/e2etest/pkg/metallb"
+	metallbv1beta1 "go.universe.tf/metallb/api/v1beta1"
+	metallbv1beta2 "go.universe.tf/metallb/api/v1beta2"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+var FRRImage string
+var _ = ginkgo.Describe("IPV4 FRRK8S-MODE Unnumbered BGP", func() {
+	var (
+		testNamespace         string
+		nodeWithP2PConnection corev1.Node
+		externalP2PContainer  *frrcontainer.FRR
+		prefixSend            = []string{"5.5.5.5/32"} //, "5555::1/128"}
+
+		cs clientset.Interface
+	)
+
+	ginkgo.BeforeEach(func() {
+		cs = k8sclient.New()
+		allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		nodeWithP2PConnection = allNodes.Items[0]
+
+		err = k8s.DeleteNamespace(k8sclient.New(), "unnumbered-bgp")
+		Expect(err).NotTo(HaveOccurred())
+		testNamespace, err = k8s.CreateTestNamespace(k8sclient.New(), "unnumbered-bgp")
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			err := k8s.DeleteNamespace(k8sclient.New(), testNamespace)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentSpecReport().Failed() {
+			k8s.DumpInfo(Reporter, ginkgo.CurrentSpecReport().LeafNodeText)
+			dumpBGPPeers(ReportPath, ginkgo.CurrentSpecReport().LeafNodeText, []*frrcontainer.FRR{externalP2PContainer})
+		}
+	})
+
+	ginkgo.Context("FRR when applying unnumbered config", func() {
+		ginkgo.It("Sessions should be established", func() {
+			rc := frrconfig.RouterConfigUnnumbered{
+				ASNLocal:      metalLBASN,
+				ASNRemote:     metalLBASN,
+				Hostname:      "tor1",
+				Interface:     "net1",
+				RouterID:      "1.1.1.1",
+				ToAdvertiseV4: []string{"1.1.1.0/24"},
+				ToAdvertiseV6: []string{"1111::/64"},
```

Comment:
does this do something?

Reply:
Was supposed to be part of the route learning but removed.